azkaban-common/src/main/java/azkaban/project/ProjectManager.java

```diff
@@ -204,6 +204,34 @@ public Project getProject(int id) {
     return projectsById.get(id);
   }
 
+    /**
+     * fetch inactive project from db by project_id
+     *
+     * @param name
+     * @return
+     */
+    public Project getInactiveProject(String name) {
```

Comment:
function name is misleading, actually the function here will fetch a project with a successful name match, no matter it is active or not .  Simple fix here is to check the project and make sure it is inactive before returning, therefore the function works as it's name indicates. 
Same comment for the function below, which gets the project by Id.